### DISCUSSION TITLE
emailRedirectTo not working with supabaseAdmin for unconfirmed user

### DISCUSSION THREAD

**Discussion by bennik88** [supabase/supabase#28091] [Questions]
# Problem:
My `emailRedirectTo` is working perfectly for users who already confirmed their email. The received mail contains the expected {{ .RedirectTo }} URL.
```js
const { data, error } = await supabase.auth.signInWithOtp({
    email: values.email,
    options: {
      emailRedirectTo: 'http://localhost:3000',
      shouldCreateUser: false,
    },
});
// Magic Link Email template
<p><a href="{{ .RedirectTo }}/auth/confirm?token_hash={{ .TokenHash }}&type=magiclink">Log In</a></p>
```
But when I create a user with `supabaseAdmin.auth.admin.createUser({email: signUpFormValues.email})`and then manually sending an email with the admin client, it always defaults to my site URL instead of the redirect URL. The URL is set in the allowed redirect settings (http://localhost:3000/**) and it is the same as when I'm redirecting an already confirmed user. I'm unable to debug it as everything is looking great in the logs but the email received from supabase contains the default siteURL instead of the redirectURL.
```js
const { data: signUpData, error: signUpError } =
      await supabaseAdmin.auth.signInWithOtp({
        email: signUpFormValues.email,
        options: {
          emailRedirectTo:'http://localhost:3000',
          shouldCreateUser: false,
        },
});
// Signup Email template
<p><a href="{{ .RedirectTo }}/auth/confirm?token_hash={{ .TokenHash }}&type=signup">Confirm your mail</a></p>
```

# What am I looking for:
I want to have my redirectURL inside of my signup email template href ('http://localhost:3000') instead of my siteUrl

**Comment by theconflictedfool** [comment]
Hey, I see that has been a problem for some time. 

Any updtes or workarounds? Currently I am just using javascript to check if the access token in present in the url, and the redirects the user to the page which requires auth.

**Comment by chikurin43** [comment]
I'm really struggling with this issue too. Is there any solution for this now?
